### ISSUE TITLE
__timeFilter

### ISSUE THREAD

**Issue by jaatsolutions** [grafana/grafana#15557]
Please include this information:
### What Grafana version are you using?
5.4.3
### What datasource are you using?
mysql 5.7.23 database
mysql plugin 5.0.0
### What OS are you running grafana on?
ubuntu 1604 lts
### What did you do?
the __timeFilter is not sending the UTC time to the server but rather the browser timezone which is causing the time values to be truncated and misrepresented.
### What was the expected result?
based on all the reading and documents we have read, we understood that all dates for grafana respect UTC.   The database datetime columns being referenced are in UTC.  

### What happened instead?
 The __timeFilter is sending the from_unixtime() in local browser datetime and NOT UTC.

### If related to metric query / data viz:

Metric as keyed in.
SELECT f.chk_date_sys AS time_sec
, f.health_class as metric
, f.health_percent as value

from F_health_check f
where f.customer = 'SC'
and f.health_code = 'FS_USED'
-- AND $__timeFilter(f.chk_date_sys)
	and f.system = 'TORBACK'
	and f.chk_date_sys BETWEEN $__timeFrom() + INTERVAL $Offset HOUR AND $__timeTo() + INTERVAL $Offset HOUR

Below is a copy of the generated SQL with the comment -- of what we want to use.   I had to use the hack to use get the offset which will NOT work for we have users in many timezones and the hack only works a the offset for the database and NOT the actual user.

SELECT f.chk_date_sys AS time_sec
, f.health_class as metric
, f.health_percent as value

from F_health_check f
where f.customer = 'SC'
and f.health_code = 'FS_USED'
-- AND f.chk_date_sys BETWEEN FROM_UNIXTIME(1550689167) AND FROM_UNIXTIME(1550699967)
	and f.system = 'TORBACK'
	and f.chk_date_sys BETWEEN FROM_UNIXTIME(1550689167) + INTERVAL 5 HOUR AND FROM_UNIXTIME(1550699967) + INTERVAL 5 HOUR

### Comment 

I had done lots of reading from the forums and noticing lots of back and forth of this working and then not working so now I am totally confused. see ticket #11619 where it says it is fixed in 5.3 but in 5.4.3 it is not working as describe in other issues.

The bottom line is the data is store in UTC.   All the panels clearly respect the converting of UTC to browser time for the actual data points. However the __timefilter and timeFrom and the __timeTo seems to not be sending UTC values but rather browser local time.

**Comment by torkelo** [comment]
They are sending in UTC, can you show the query grafana generated and your current local browser time / UTC time?

**Comment by jaatsolutions** [comment]
As requested here is the output the clearly shows the macro are NOT sending UTC time but rather local time.

Ran this test at about 8:15 ontario/toronto time.   What other information do you need.   Again, I have been working this issue for days and the hack used will work but it is not sustainable as we move forward.   What am I doing wrong.   Database datetime column is IN UTC for all our data warehouse tables transnational data is in UTC.   The server is in ontario/toronto.  

![image](https://user-images.githubusercontent.com/47832743/53171667-6a468200-35b1-11e9-8fa7-f314e4317708.png)


Here is the metric code and the generated SQL

![image](https://user-images.githubusercontent.com/47832743/53171879-ee990500-35b1-11e9-85ad-d1cd4d373691.png)

**Comment by torkelo** [comment]
It sending local time as UTC , which is correct, please show your browser time , the dashboard time range and the generated sql.

If this was a bug in Grafana the many hundreds would report this so it’s likely that it’s something unique about your setup. 

But it could be a grafana but so please continue to investigate and send us more I for

**Comment by jaatsolutions** [comment]
OK, I may NOT be explaining properly.   The pictures show in fact it is NOT sending UTC but rather local.

The columns from macro and to macro are what grafana is generating.

The browser was in ontario/toronto time -5 from UTC.   If you look at the sql generated I provided insight into what the sql that was generated.

So the sql was generated at 8:15 EST the value it sent was 8:15 for UTC for the __timeTo and it should have been 13:15 for __timeTo.

I think people are talking about this for there is lots of back and forth of mysql plugin working and then not working.

Further proof if you look at the code I sent I also used the __timeFrom and __timeTo using the timezone offset to get the results back.   So I can make the panel work if I add the offset which only proves it did not see UTC.

First image shows the sql that was generated by the grafana interface

Second image shows what I used as the metric sql, bottom have is what grafana interface 'generated sql' Produced.

The time value is NOT showing UTC but local.

browser again is set to ontario/toronto (EST) which is -5 currently.

hope this helps.

The dasboard range was last 6 hours.   Again, the sql demonstrates that for I used __timeFrom and __timeTo along with __timeFilter

**Comment by torkelo** [comment]
When I check timeTo:  1550755252*1000, it's 2019-02-21T13:20:52Z UTC
timeFrom: 1550733652*1000 = 2019-02-21T07:20:52Z

It's strange, when I check __timeFilter generates correct UTC epochs. 

maybe you need to change your MySQL server to UTC? MySQL FROM_UNIXTIME will take the unix UTC epoch and convert it to the MySQL server timezone

**Comment by jaatsolutions** [comment]
Moving the mysql cluster to UTC is problematic for many reasons for it is the workhorse of our company and many applications use the infrastructure beyond just our data warehouse.

As you can see we built the panel correctly and the mysql plugin is generating the wrong value for the macro.

Is there some other method to get the mysql datasource plugin to respond correctly.

**Comment by jaatsolutions** [comment]
We have changed the database servers in the cluster to UTC.   This has fixed the issue for grafana but has created a lot of issues for the data warehouse.

So I do not think this issue is resolved for it does not meet with the criteria that all things grafana are utc when it fact it is not.   

I would be curious to know what other customers using mysql and grafana are experiencing this issue.

**Comment by torkelo** [comment]
But everything is in UTC, it’s your MySQL that is doing the local translation

**Comment by jaatsolutions** [comment]
I disagree.   You should clarify your documentation to state the __timeFormat will convert the panel dates to your database server local time.   That is a truth.   but to state that it converts to UTC is a falsehood.

This clarification in your documentation would have saved me and my team a lot of frustration.

**Comment by torkelo** [comment]
Your right the documentation can be clearer and I am sorry this cause confusion and time wasted. We found that MySQL date management and timezone to be really poor, basically MySQL always forces dates to be interpolated to local time (server timezone).
